### PR TITLE
ConfigContext data migration and make sure the SiteLocationType.content_types is set to allow all Location ContentTypes

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: N/A
# What's Changed

1. Migrate ConfigContext regions and sites data to ConfigContext.locations.
2. Ensure the new "Site" LocationType allows all ContentTypes allowed by Location (FeatureQuery("locations").get_query()).
3. Tested thoroughly and did what it was intended.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py [hunk 1]
```diff
@@ -175,23 +176,48 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
             exclude_lt="Site",
         )
 
+    # Reassign Region Models to Locations of Region LocationType
+    if Region.objects.exists():
+        ConfigContext = apps.get_model("extras", "configcontext")
+        ccs = ConfigContext.objects.filter(regions__isnull=False).prefetch_related("locations", "regions")
+        for cc in ccs:
+            region_name_list = list(cc.regions.all().values_list("name", flat=True))
+            region_locs = list(Location.objects.filter(name__in=region_name_list, location_type=region_lt))
```
Comment (glennmatthews) [inline nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py:185]: Wonder if we should print a warning if `region_locs.count() < len(region_name_list)`?

File: nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py [hunk 2]
```diff
@@ -175,23 +176,48 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
...
+
         cts = CircuitTermination.objects.filter(location__isnull=True).select_related("site")
         for ct in cts:
             ct.location = Location.objects.get(name=ct.site.name, location_type=site_lt)
         CircuitTermination.objects.bulk_update(cts, ["location"], 1000)
 
+        ccs = ConfigContext.objects.filter(sites__isnull=False).prefetch_related("locations", "sites")
+        for cc in ccs:
+            site_name_list = list(cc.sites.all().values_list("name", flat=True))
+            site_locs = list(Location.objects.filter(name__in=site_name_list, location_type=site_lt))
```
Comment (glennmatthews) [inline nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py:215]: Same thought here.

File: nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py [hunk 3]
```diff
@@ -175,23 +176,48 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
...
+            original_locs = list(cc.locations.all())
+            locations = site_locs + original_locs
+            cc.locations.set(site_locs + original_locs)
```
Comment (glennmatthews) [inline nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py:218]: Similarly?

```suggestion
            cc.locations.add(site_locs)
```

**Review by glennmatthews — APPROVED** [review]

File: nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py [hunk 4]
```diff
@@ -175,23 +176,48 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
...
+    # Reassign Region Models to Locations of Region LocationType
+    if Region.objects.exists():
+        ConfigContext = apps.get_model("extras", "configcontext")
+        ccs = ConfigContext.objects.filter(regions__isnull=False).prefetch_related("locations", "regions")
+        for cc in ccs:
+            region_name_list = list(cc.regions.all().values_list("name", flat=True))
+            region_locs = list(Location.objects.filter(name__in=region_name_list, location_type=region_lt))
+            original_locs = list(cc.locations.all())
+            locations = region_locs + original_locs
+            cc.locations.set(locations)
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py:188]: `.add()` does not work well with lists or querysets apparently.

```
>>> ConfigContext.objects.first()
<ConfigContext: 123>
>>> c = ConfigContext.objects.first()
>>> c.locations.add(Location.objects.all()
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/__init__.py", line 2434, in to_python
    return uuid.UUID(**{input_form: value})
  File "/usr/local/lib/python3.7/uuid.py", line 157, in __init__
    hex = hex.replace('urn:', '').replace('uuid:', '')
AttributeError: 'LocationQuerySet' object has no attribute 'replace'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/related_descriptors.py", line 952, in add
    through_defaults=through_defaults,
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/related_descriptors.py", line 1126, in _add_items
    target_ids = self._get_target_ids(target_field_name, objs)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/related_descriptors.py", line 1073, in _get_target_ids
    target_ids.add(target_field.get_prep_value(obj))
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/related.py", line 977, in get_prep_value
    return self.target_field.get_prep_value(value)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/__init__.py", line 2418, in get_prep_value
    return self.to_python(value)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/__init__.py", line 2439, in to_python
    params={'value': value},
django.core.exceptions.ValidationError: ['“<LocationQuerySet [<Location: Campus-00>, <Location: Building-01>, <Location: Elevator-02>, <Location: Elevator-09>, <Location: Floor-03>, <Location: Room-04>, <Location: Aisle-05>, <Location: Aisle-12>, <Location: Room-11>, <Location: Aisle-19>, <Location: Campus-07>, <Location: Building-08>, <Location: Floor-10>, <Location: Room-18>, <Location: Campus-14>, <Location: Building-15>, <Location: Elevator-16>, <Location: Floor-17>, <Location: Root-06>, <Location: Root-13>]>” is not a valid UUID.']
>>> c.locations.add([Location.objects.all()])
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/db/models/fields/__init__.py", line 2434, in to_python
    return uuid.UUID(**{input_form: value})
  File "/usr/local/lib/python3.7/uuid.py", line 157, in __init__
    hex = hex.replace('urn:', '').replace('uuid:', '')
AttributeError: 'list' object has no attribute 'replace'

During handling of the above exception, another exception occurred:
```

**Comment by HanlinMiao** [comment]
See https://github.com/nautobot/nautobot/pull/3208#discussion_r1095829166 for more migrations we need to do in this PR

File: nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jathanism) [reply nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py:188]: You have to unpack the list e.g. `cc.locations.add(*region_locs)`

File: nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py (hunk 3 above)
Comment (jathanism) [reply nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py:218]: ^ I edited your suggestion to unpack the list.

**Pull request merged**